src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -170,9 +170,82 @@ private static bool ApplyChangesFromTC35(JObject sarifLog)
 
             modifiedLog |= RenameFixChangesToFixArtifactChanges(sarifLog);
 
+            modifiedLog |= RenameArtifactRolesEnums(sarifLog);
+
             return modifiedLog;
         }
 
+        private static bool RenameArtifactRolesEnums(JObject sarifLog)
+        {
+            string[] artifactRolesPathsToUpdate =
+            {
+                "inlineExternalProperties[].artifacts[]",
+                "runs[].artifacts[]"
+            };
+
+            bool actionOnLeafNode(JObject artifact)
+            {
+                if (artifact["roles"] is JArray roles)
+                {
+                    bool isModified = false;
+
+                    foreach(JValue role in roles)
+                    {
+                        switch (role.Value as string)
+                        {
+                            case "unmodifiedFile":
+                                {
+                                    role.Value = "unmodified";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "modifiedFile":
+                                {
+                                    role.Value = "modified";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "addedFile":
+                                {
+                                    role.Value = "added";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "deletedFile":
+                                {
+                                    role.Value = "deleted";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "renamedFile":
+                                {
+                                    role.Value = "renamed";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "uncontrolledFile":
```

Comment:
>uncontrolledFile [](start = 34, length = 16)

you could have put all of these in a single case that rename the item simply by trimming the original file by its "File" suffix.

Reply:
oh! correct! thanks!

---
In reply to: [277023544](https://github.com/Microsoft/sarif-sdk/pull/1439#discussion_r277023544) [](ancestors = 277023544)